x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/check_error_details.ts

```diff
@@ -18,9 +23,20 @@ export const checkErrorDetails = (error: unknown): { isUserError: boolean } => {
     return { isUserError: true };
   }
 
+  if (
+    (error instanceof Error &&
+      typeof error.message === 'string' &&
+      (error.message as string).endsWith('missing')) ||
+    (typeof error === 'string' && error.endsWith('missing') && error.split(' ').length === 2)
```

Comment:
checking length here because the message returned from ES is seemingly always `<job name> missing`. So to ensure we are not checking "missing" as a user error for other possible error messages, I added this check.

Reply:
Given that this is now part of this long boolean expression, I would consider capturing it as e.g. `isMlJobMissingError()` so that it doesn't get accidentally refactored later on. It's fine as it is, though.